### PR TITLE
agent: support changing user of the new connection

### PR DESCRIPTION
**Pull request by diogofcunha**

### CODE REVIEW

File: tests/AgentTests/SQLiteTests.hs [hunk 1]
```diff
@@ -336,6 +342,81 @@ testGetRcvConn =
...
+    connResult <- getConn db connId
+
+    case connResult of
+      Right (SomeConn SCNew (NewConnection connData)) -> do
+        let ConnData {userId} = connData
+        userId `shouldBe` newUserId
+      _ -> do
+         expectationFailure "Failed to get connection"
+
+testSetConnUserIdNewConnMulti :: SpecWith SQLiteStore
```
Comment (spaced4ndy) [inline tests/AgentTests/SQLiteTests.hs:361]: for future better to add tests to FunctionalAPITests instead - here they may be harder to maintain

File: tests/AgentTests/SQLiteTests.hs [hunk 2]
```diff
@@ -336,6 +342,81 @@ testGetRcvConn =
...
+    _ <- setConnUserId db 1 connId newUserId
+    _ <- setConnUserId db newUserId connId 1
+    connResult <- getConn db connId
+
+    case connResult of
+      Right (SomeConn SCNew (NewConnection connData)) -> do
+        let ConnData {userId} = connData
+        userId `shouldBe` 1
+      _ -> do
+         expectationFailure "Failed to get connection"
```
Comment (spaced4ndy) [inline tests/AgentTests/SQLiteTests.hs:376]:
```suggestion
      _ -> expectationFailure "Failed to get connection"
```

Comment (spaced4ndy) [inline tests/AgentTests/SQLiteTests.hs:376]:
```suggestion
    getConn db connId >>= \case
      Right (SomeConn SCNew (NewConnection ConnData {userId})) ->
        userId `shouldBe` 1
      _ -> expectationFailure "Failed to get connection"
```

Comment (spaced4ndy) [reply tests/AgentTests/SQLiteTests.hs:376]: also same in similar places

File: src/Simplex/Messaging/Agent.hs [hunk 3]
```diff
@@ -334,6 +335,11 @@ createConnection :: AgentClient -> UserId -> Bool -> SConnectionMode c -> Maybe
 createConnection c userId enableNtfs = withAgentEnv c .:: newConn c userId "" enableNtfs
 {-# INLINE createConnection #-}
 
+-- | Changes the user id associated with a connection 
+updateConnectionUserId :: AgentClient -> UserId -> ConnId -> UserId -> AE ()
+updateConnectionUserId c oldUserId connId newUserId = withAgentEnv c $ updateConnUserId c oldUserId connId newUserId
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:340]: this is the first case of using `withAgentEnv` - it shouldn't be used here

File: src/Simplex/Messaging/Agent/Store/SQLite.hs [hunk 4]
```diff
@@ -1976,6 +1977,17 @@ setConnDeleted db waitDelivery connId
   | otherwise =
       DB.execute db "UPDATE connections SET deleted = ? WHERE conn_id = ?" (True, connId)
 
+setConnUserId :: DB.Connection -> UserId -> ConnId -> UserId -> IO (Either StoreError ())
+setConnUserId db oldUserId connId newUserId = getConn db connId $>>= \case
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent/Store/SQLite.hs:1981]:
```suggestion
setConnUserId db oldUserId connId newUserId =
  getConn db connId $>>= \case
```

File: tests/AgentTests/SQLiteTests.hs [hunk 5]
```diff
@@ -114,6 +114,12 @@ storeTests = do
           testDeleteRcvConn
           testDeleteSndConn
           testDeleteDuplexConn
+        describe "setConnUserId" $ do
+          testSetConnUserIdNewConn
+          testSetConnUserIdNewConnMulti
+          testSetConnUserIdRcvConn
+          testSetConnUserIdSndConn
+          testSetConnUserIdDuplexConn
```
Comment (epoberezkin) [inline tests/AgentTests/SQLiteTests.hs:122]:
```suggestion
```

File: src/Simplex/Messaging/Agent/Store/SQLite.hs [hunk 6]
```diff
@@ -1976,6 +1977,17 @@ setConnDeleted db waitDelivery connId
...
+    (SomeConn _ NewConnection {}) -> updateConn
+    (SomeConn _ RcvConnection {}) -> updateConn
+    (SomeConn c _) -> pure . Left . SEBadConnType $ connType c
+  where
+    updateConn :: IO (Either StoreError ())
+    updateConn = do 
+      DB.execute db "UPDATE connections SET user_id = ? WHERE conn_id = ? and user_id = ?" (newUserId, connId, oldUserId)
+      pure $ Right ()
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent/Store/SQLite.hs:1989]:
```suggestion
    updateConn :: IO()
    updateConn =
      DB.execute db "UPDATE connections SET user_id = ? WHERE conn_id = ? and user_id = ?" (newUserId, connId, oldUserId)
```

File: src/Simplex/Messaging/Agent/Store/SQLite.hs (hunk 4 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Agent/Store/SQLite.hs:1981]: this boundary is usually in a different file

**Pull request merged**